Source/diffusion/Castro_diffusion.cpp

```diff
@@ -5,9 +5,61 @@
 using std::string;
 
 #include "Diffusion.H"
+#include "eos.H"
+#include "conductivity.H"
 
 using namespace amrex;
 
+void
+Castro::fill_temp_cond(const Box& bx,
+                       Array4<Real const> const U_arr,
+                       Array4<Real> const coeff_arr) {
+
+  Real lsmall_temp = small_temp;
+  Real ldiffuse_cutoff_density = diffuse_cutoff_density;
+  Real ldiffuse_cutoff_density_hi = diffuse_cutoff_density_hi;
+
+  AMREX_PARALLEL_FOR_3D(bx, i, j, k,
+  {
+
+    eos_t eos_state;
+    eos_state.rho  = U_arr(i,j,k,URHO);
+    Real rhoinv = 1.0_rt/eos_state.rho;
+
+    eos_state.T = U_arr(i,j,k,UTEMP);   // needed as an initial guess
+    eos_state.e = U_arr(i,j,k,UEINT) * rhoinv;
+    for (int n = 0; n < NumSpec; n++) {
+      eos_state.xn[n] = U_arr(i,j,k,UFS+n) * rhoinv;
+    }
+    for (int n = 0; n < NumAux; n++) {
+      eos_state.aux[n] = U_arr(i,j,k,UFX+n) * rhoinv;
+    }
+
+    if (eos_state.e < 0.0_rt) {
+      eos_state.T = lsmall_temp;
+      eos(eos_input_rt, eos_state);
+    } else {
+      eos(eos_input_re, eos_state);
+    }
+
+
+    if (eos_state.rho > ldiffuse_cutoff_density) {
+      conductivity(eos_state);
+
+      if (eos_state.rho < diffuse_cutoff_density_hi) {
```

Comment:
We need local copies of `diffuse_cutoff_density_hi` and `diffuse_cond_scale_fac`.